core/lib/config/src/configs/contracts.rs

```diff
@@ -37,6 +37,8 @@ pub struct ContractsConfig {
     pub l2_testnet_paymaster_addr: Option<Address>,
     pub l1_multicall3_addr: Address,
     pub ecosystem_contracts: Option<EcosystemContracts>,
+    // Used by the RPC API and by the node builder in wiring the BaseTokenRatioProvider layer. If not set,
+    // ETH is the chain's base token and the "ETH address" is SHARED_BRIDGE_ETHER_TOKEN_ADDRESS.
     pub base_token_addr: Option<Address>,
```

Comment:
Nit: you may add a `pub fn base_token_addr() -> Address { self.base_token_addr.unwrap_or(SHARED_BRIDGE_ETHER_TOKEN_ADDRESS) }` to simplify usage.

Reply:
I'm not so convinced. The `base_token_addr` being set to ETH address happens during chain registration. If the chain registration logic changes and we keep this field optional we may hide a bug. This was why I wanted to remove the optionality. I think that my clarification comment has done more to confuse than clarify so removing it!

Reply:
To not hide a bug, we may add a cross-check to make sure that the variable has the same value as contract (like [here](https://github.com/matter-labs/zksync-era/blob/main/core/node/node_framework/src/implementations/layers/l1_batch_commitment_mode_validation.rs)), and it would be a better guarantee.
We're only talking about convenience here, not safety.